### ISSUE TITLE
A plea for open-source spacecraft & asteroid models (some existing models are broken in GLES2 web export)

### ISSUE THREAD

**Issue by charliewhitfield** [ivoyager/ivoyager#11]
Graphic artists, I need your help!

I have Juno in the Planetarium, but only as an orbit without model. The existing [open-source model I found](https://solarsystem.nasa.gov/resources/2376/juno-3d-model/) crashes in GLES2 web export (which is what we have to use for the Planetarium, at least for now).

Someone posted an open-source JWST model online but then it disappeared.

Other priorities: Voyager 1 & 2, New Horizons, Tiangong, or really anything from the real world that you'd like to see in our simulator (Elon Musk's roadster, if you like...).

Perhaps this is too big an ask, but it would really help if we had full and reduced size assets (the latter to help in web deployment).

I'm working on a celestial navigation system (patched conics) to show the historical flight paths for non-orbiting probes. So that's why I'd really like to have the Voyagers and New Horizons models...

**Comment by Zireael07** [comment]
Doubt you will find open source models of real life stations/telescopes, as they're likely copyrighted.

As for asteroids, there's a couple random 3d models around (OGA and similar sites)

**Comment by charliewhitfield** [comment]
@Zireael07,

Yep, I got a few of those in already. (Edited top post to indicate what we **do** have.) Please let me know if you find others!

**Comment by TheLostProbe** [comment]
[Celestia](https://github.com/CelestiaProject/Celestia) is FOSS, and includes models of most of the spacecraft (and asteroids) that you're looking for. you can get them from the [CelestiaContent repository](https://github.com/celestiaProject/CelestiaContent)
even more community-created models, as well as ones that improve Celestia's default models, are available [here](https://celestia.mobi/resources/categories?parent=22FD36EF-8F97-5C31-30FE-A76257FC2DE1), [here](https://celestiaproject.space/forum/viewtopic.php?f=23&t=18024), and [here](https://celestiaproject.space/forum/viewtopic.php?f=23&t=23836)
more models for minor planets (asteroids, comets, TNOs, centaurs etc) can be obtained from [here](https://celestia.mobi/resources/category?category=5DABA8EB-7105-8592-2DA8-505AC6584A55), [here](https://celestia.mobi/resources/categories?parent=C789D7FC-F80E-EBDA-94BA-79FBE31FF9C1), and [here](https://celestia.mobi/resources/category?category=4F4D674A-6BBE-EC82-3F1A-9CBA4E91044D)

quite a few of these are very heavy models, though, and the textures don't help either. I'm not sure how they could be optimized, but I hope this gives you a starting point and I'm interested to see where this project goes next

**Comment by charliewhitfield** [comment]
Thanks @TheLostProbe!

Looks like the minor-planets are a mix of visited (i.e., actually imaged) and non-visited (fictional representations). Fictional representations are ok but I'll want to treat them differently. Maybe they can be in the game platform but not in the Planetarium. Or maybe in the Planetarium as a toggle option. In any case I'll want to prioritize visited minor planets (along with particular spacecrafts).

I'll take a deeper look after I get our Godot 4 migration fully working.

It's going to be a while however before I'm able to take on model conversions. I'd be overjoyed if someone else wants to take that on before I get to it...

(Seriously, I'd like to be able to add a contributors section to the author's page maybe by the end of the year...)

**Comment by TheLostProbe** [comment]
well, Anim8or should be able to run on my crappy computer, so maybe I could convert them for you? just let me know what format you want them exported in and I'll get to work

**Comment by charliewhitfield** [comment]
Great!

Here's the [Godot supported formats](https://docs.godotengine.org/en/stable/tutorials/assets_pipeline/importing_scenes.html#doc-importing-3d-scenes). I think all our current models are .glb, so that might be slightly preferred. I'll be happy as long as Godot can import it. After import, it's in Godot binary format regardless, so original format doesn't matter too much.

Try to prioritize real-image [visited minor planets](https://en.wikipedia.org/wiki/List_of_minor_planets_and_comets_visited_by_spacecraft) first.

For spacecrafts, the ones I can add very quickly are the planetary probes that are (or were) orbiting something for a while, or are at a Lagrange point. I really do want to create navigation paths for Voyager 1, 2, New Horizons, etc., but that will take some work...

**Comment by TheLostProbe** [comment]
> I think all our current models are .glb, so that might be slightly preferred. I'll be happy as long as Godot can import it.

if Anim8or doesn't export GLBs, I can probably just export them as DAE or OBJ and then convert them over to GLB
> Try to prioritize real-image [visited minor planets](https://en.wikipedia.org/wiki/List_of_minor_planets_and_comets_visited_by_spacecraft) first.

of course!
> For spacecrafts, the ones I can add very quickly are the planetary probes that are (or were) orbiting something for a while, or are at a Lagrange point. I really do want to create navigation paths for Voyager 1, 2, New Horizons, etc., but that will take some work...

if I, Voyager supports NASA's SPICE kernels then I should be able to fish around for some SPICE trajectories of interplanetary probes for you. if not, then I guess SPICE support can be added to the roadmap ;)
> Be sure to record attribution information and see if you can find the license for each asset. It has to be truly open-source for us to distribute it.

sure, I'll see if I can contact the creators of the add-ons and ask where they got their models from. I'm fairly certain that they are open-source, but better safe than sorry
